src/app/components/instructions-voice/instructions-voice.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { navActionTypes } from '../../store/nav.actions';
+import { tap, withLatestFrom } from 'rxjs/operators';
+import { getNavState } from '../../store/nav.selectors';
+
+@Component({
+  selector: 'app-instructions-voice',
+  templateUrl: './instructions-voice.component.html',
+  styleUrls: ['./instructions-voice.component.scss']
+})
+export class InstructionsVoiceComponent implements OnInit {
+  @Effect({ dispatch: false })
+  playVoiceInstruction$ = this.actions$.pipe(
+    ofType(navActionTypes.PLAY_VOICE_WP_NOTIFICATION),
+    withLatestFrom(this.store.pipe(select(getNavState))),
+    tap(([playVoiceAction, stateData]) => {
+      const instructionText = `in ${(<any>playVoiceAction).payload.distanceNotification} meters
+       ${stateData.routeDetails.routeLegs[stateData.nextWaypointIndex].maneuverType} to
```

Comment:
Maneuver type should't be played as is we should have a map from api maneuver type to instruction.
Please add as a new issue